src/annotator/sidebar.js

```diff
@@ -0,0 +1,326 @@
+        frameVisibleWidth -= leftMargin;
+      } else {
+        frameVisibleWidth += width;
+      }
+
+      // Infer expanded state based on whether at least part of the sidebar
+      // frame is visible.
+      expanded = frameVisibleWidth > toolbarWidth;
+    }
+
+    this.onLayoutChange({
+      expanded,
+      width: expanded ? frameVisibleWidth : toolbarWidth,
+      height: rect.height,
+    });
+  }
+
+  _onPan(event) {
+    if (!this.frame) {
+      return;
+    }
+
+    switch (event.type) {
+      case 'panstart':
+        this._resetGestureState();
+
+        // Disable animated transition of sidebar position
+        this.frame.addClass('annotator-no-transition');
+
+        // Disable pointer events on the iframe.
+        this.frame.css('pointer-events', 'none');
+
+        this._gestureState.initial = parseInt(
+          getComputedStyle(this.frame[0]).marginLeft
+        );
+
+        break;
+      case 'panend':
+        this.frame.removeClass('annotator-no-transition');
+
+        // Re-enable pointer events on the iframe.
+        this.frame.css('pointer-events', '');
+
+        // Snap open or closed.
+        if (
+          this._gestureState.final === null ||
+          this._gestureState.final <= -MIN_RESIZE
+        ) {
+          this.show();
+        } else {
+          this.hide();
+        }
+        this._resetGestureState();
+        break;
+      case 'panleft':
+      case 'panright': {
+        if (typeof this._gestureState.initial !== 'number') {
+          return;
+        }
+
+        const margin = this._gestureState.initial;
+        const delta = event.deltaX;
+        this._gestureState.final = Math.min(Math.round(margin + delta), 0);
+        this._updateLayout();
+        break;
+      }
+    }
+  }
+
+  _onSwipe(event) {
+    switch (event.type) {
+      case 'swipeleft':
+        this.show();
+        break;
+      case 'swiperight':
+        this.hide();
+        break;
+    }
+  }
+
+  show() {
+    this.crossframe.call('sidebarOpened');
+
+    if (this.frame) {
+      this.frame.css('margin-left', `${-1 * this.frame.width()}px`);
+      this.frame.removeClass('annotator-collapsed');
+    }
+
+    this.toolbar.sidebarOpen = true;
+
+    if (this.options.showHighlights === 'whenSidebarOpen') {
+      this.setVisibleHighlights(true);
+    }
+
+    this._notifyOfLayoutChange(true);
+  }
+
+  hide() {
+    if (this.frame) {
+      this.frame.css('margin-left', '');
+      this.frame.addClass('annotator-collapsed');
+    }
+
+    this.toolbar.sidebarOpen = false;
+
+    if (this.options.showHighlights === 'whenSidebarOpen') {
+      this.setVisibleHighlights(false);
+    }
+
+    this._notifyOfLayoutChange(false);
+  }
+
+  isOpen() {
+    if (this.frame) {
+      return !this.frame.hasClass('annotator-collapsed');
+    } else {
+      // Assume an external frame is always open.
+      return true;
+    }
+  }
+
+  setAllVisibleHighlights(shouldShowHighlights) {
```

Comment:
What does this function accomplish (in a conceptual way)?

Reply:
It turns the document highlights on or off. I've added a JSDoc comment.